frontend/src/components/file-explorer/types.d.ts

```diff
@@ -0,0 +1,4 @@
+interface TreeNode {
+  name: string;
+  children?: TreeNode[];
+}
```

Comment:
Why do we need this extra type? We basically have the same thing in `fileService`